### PR TITLE
CacheConfigReadOnly test

### PR DESCRIPTION
**Pull request by mustafaiman**
adds missing UnsupportedOperation tests

### CODE REVIEW

**Comment by serkan-ozal** [comment]
:+1:

File: hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java [hunk 1]
```diff
@@ -34,7 +34,42 @@
...
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void settingHotRestartEnabledOfReadOnlyCacheConfigShouldFail() {
+        getCacheConfigReadOnly().setHotRestartEnabled(false);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void settingPartitionLostListenerConfigsOnReadOnlyCacheConfigShouldFail() {
+        getCacheConfigReadOnly().setPartitionLostListenerConfigs(null);
```
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java]: seems we already have this test: https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java#L150

**Comment by serkan-ozal** [comment]
verify

File: hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java (hunk 1 above)
Comment (mustafaiman) [inline hazelcast/src/test/java/com/hazelcast/cache/config/CacheConfigReadOnlyTest.java]: These two are different classes; `CacheConfigReadOnly` and `CacheSimpleConfigReadOnly`.

**Comment by mustafaiman** [comment]
verify

**Pull request merged**